File: src/utils/money.ts
/** 格式化金额显示 */
export function formatMoney(value: number) {
  const n = Number(value) || 0
  return n.toLocaleString('zh-CN', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  })
}

/** 带正负号的金额文案 */
export function signedMoney(type: 'income' | 'expense', value: number) {
  const prefix = type === 'income' ? '+' : '-'
  return `${prefix}${formatMoney(value)}`
}

File: lib/core/utils/money.dart
import 'package:intl/intl.dart';

final _moneyFmt = NumberFormat('#,##0.00', 'zh_CN');

/// 格式化金额显示。
String formatMoney(num value) => _moneyFmt.format(value);

/// 带正负号的金额文案。
String signedMoney(String type, num value) {
  final prefix = type == 'income' ? '+' : '-';
  return '$prefix${formatMoney(value)}';
}
